test/GasAccounting.t.sol

```diff
@@ -0,0 +1,318 @@
+        _initializeEscrowLock(executionEnvironment, gasMarker, userOpValue);
+    }
+
+    function assign(address owner, uint256 value) external returns (bool) {
+        return _assign(owner, value);
+    }
+
+    function credit(address owner, uint256 value) external {
+        _credit(owner, value);
+    }
+
+    function trySolverLock(SolverOperation calldata solverOp) external returns (bool) {
+        return _trySolverLock(solverOp);
+    }
+
+    function releaseSolverLock(SolverOperation calldata solverOp, uint256 gasWaterMark, uint256 result) external {
+        _releaseSolverLock(solverOp, gasWaterMark, result);
+    }
+
+    function settle(address winningSolver, address bundler) external {
+        _settle(winningSolver, bundler);
+    }
+
+    function increaseBondedBalance(address account, uint256 amount) external {
+        deal(address(this), amount);
+        accessData[account].bonded += uint128(amount);
+        bondedTotalSupply += uint128(amount);
+    }
+
+    function increaseUnbondingBalance(address account, uint256 amount) external {
+        deal(address(this), amount);
+        _balanceOf[account].unbonding += uint128(amount);
+        bondedTotalSupply += uint128(amount);
+    }
+}
+
+contract GasAccountingTest is Test {
+    MockGasAccounting public mockGasAccounting;
+    address executionEnvironment = makeAddr("executionEnvironment");
+
+    uint256 initialClaims;
+    SolverOperation solverOp;
+
+    function setUp() public {
+        mockGasAccounting = new MockGasAccounting(0, address(0), address(0));
+        uint256 gasMarker = gasleft();
+
+        mockGasAccounting.initializeEscrowLock(executionEnvironment, gasMarker, 0);
+
+        initialClaims = getInitialClaims(gasMarker);
+        solverOp.from = makeAddr("solver");
+    }
+
+    function getInitialClaims(uint256 gasMarker) public view returns (uint256 claims) {
+        uint256 rawClaims = (gasMarker + 1) * tx.gasprice;
+        claims = rawClaims + ((rawClaims * mockGasAccounting.SURCHARGE()) / mockGasAccounting.SURCHARGE_BASE());
+    }
+
+    function test_validateBalances() public {
+        assertFalse(mockGasAccounting.validateBalances());
+
+        mockGasAccounting.trySolverLock(solverOp);
+        mockGasAccounting.reconcile{ value: initialClaims }(executionEnvironment, solverOp.from, 0);
+
+        assertTrue(mockGasAccounting.validateBalances());
+    }
+
+    function test_contribute() public {
+        vm.expectRevert(
+            abi.encodeWithSelector(FastLaneErrorsEvents.InvalidExecutionEnvironment.selector, executionEnvironment)
+        );
+        mockGasAccounting.contribute();
+
+        uint256 contributeValue = 1000;
+        deal(executionEnvironment, contributeValue);
+
+        vm.prank(executionEnvironment);
+        mockGasAccounting.contribute{ value: contributeValue }();
+
+        assertEq(mockGasAccounting.deposits(), contributeValue);
```

Comment:
Should this also test the value of the contract in addition to the deposits value? Maybe also that they match?

Reply:
Added in https://github.com/FastLane-Labs/atlas/pull/88/commits/1a78f6790387bba029bb41c17776e15d6697f45c.